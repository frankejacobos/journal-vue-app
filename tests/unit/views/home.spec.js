import { shallowMount } from "@vue/test-utils";
import Home from "@/views/HomeView.vue";

describe("Pruebas en la vista Home", () => {
  test("debe de hacer match con el snapshot", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("debe redireccionar a no-entry al hacer click en un boton", () => {
    const mockRouter = { push: jest.fn() };
    const wrapper = shallowMount(Home, {
      global: { mocks: { $router: mockRouter } },
    });
    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
