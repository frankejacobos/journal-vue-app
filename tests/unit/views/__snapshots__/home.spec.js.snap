// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la vista Home debe de hacer match con el snapshot 1`] = `
<div class="home-view">
  <h1>Home</h1>
  <p>This is the Home page</p><button class="btn btn-primary">Go to DayBook</button>
</div>
`;
